guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc

```diff
@@ -0,0 +1,497 @@
+    name: Apache-2.0
+    url: "https://www.apache.org/licenses/LICENSE-2.0.html"
+tags: # <3>
+  - name: books
+    description: Search for books and add new ones
+----
+<1> The version that will be used for parsing.
+<2> The info object contains general information about the api to its reader.
+<3> Tags will be used to logically structure different paths.
+
+NOTE: If you are new to OpenAPI, you might be interested in reading the
+link:https://swagger.io/docs/specification/about/[OpenAPI guide] or the
+link:https://swagger.io/specification/[OpenAPI 3.0.0 specification] after you finish this guide.
+
+
+=== Paths and operations definition
+
+We will proceed by defining a path that will be available in our API. Paste the following to our file:
+
+[source,yaml]
+----
+paths:
+  /search:
+    get: # <1>
+      tags:
+        - books # <2>
+      summary: Search for a book
+      operationId: search # <3>
+      parameters: # <4>
+        - name: book-name
+          in: query
+          schema:
+            type: string
+            minLength: 3 # <5>
+        - name: author-name
+          in: query
+          schema:
+            type: string
+      responses: # <6>
+        "200": # <7>
+          description: Success
+          content:
+            "applicaton/json":
+              schema:
+                type: array,
+                items:
+                  $ref: "#/components/schemas/BookInfo"
+        "400": # <8>
+          description: Bad Request
+----
+<1> We define the `GET` operation on the `/search` path.
+<2> We use the `books` tag that we previously defined. Note that for each tag a Controller will be generated
+that will implement its operations.
+<3> The `search` operation id will be used as method name for the given path.
+<4> We define 2 parameters of type string that user should supply in query.
+<5> Validation can be used on parameters. In this case book name must contain at least 3 characters.
+<6> The `responses` object describes the response codes that can be produced.
+It also defines the structure of body if any.
+<7> In case of correct request we define the body to contain a list of `BookInfo` objects. The schema for the book info
+object will be defined later in `components/schemas` section of definition.
+<8> The `"400"` status code will be produced by Micronaut in case of bad request, like an incorrect type supplied or
+failed validation. Even though Micronaut handles it automatically and no implementation is needed on our side, we add it
+for a complete api specification.
+
+NOTE: You can read more about parameter descriptions in the
+link:https://swagger.io/docs/specification/describing-parameters/["Describing Parameters" OpenAPI guide].
+All the available types and their validations are described in
+link:https://swagger.io/docs/specification/data-models/data-types/["Data Models (Schemas)" OpenAPI guide].
+
+We will define another path with `POST` operation similarly to the way we created the first one:
```

Comment:
"similarly to the way" -> "like"

Reply:
Removed the subordinate clause completely ("similarly to the way we created the first one" -> "")